### PR TITLE
Performing auth model calls on controller instead of view

### PR DESCRIPTION
**Pull request by maxsolomon14**

### CODE REVIEW

File: app/Http/Controllers/MessagesController.php [hunk 1]
```diff
@@ -75,18 +75,16 @@ public function store(Request $request, $sender_id, $recipient_id)
      */
     public function show(Message $message, $sender_id, $recipient_id)
     {
-        // $users = Message::where(function ($query) use ($recipient_id, $sender_id) {
-        //     $query->where('sender_id', '=', $recipient_id)
-        //           ->orWhere('sender_id', '=', $sender_id);
-        // })->first();
         if ($recipient_id === Auth::user()->id or $sender_id === $recipient_id) {
```
Comment (alexroden) [inline app/Http/Controllers/MessagesController.php]: I would say that you are better off using `||` here, as that is the standard that many developer work by. Although, I don't think that there is a performance benefit

File: app/Http/Controllers/MessagesController.php [hunk 2]
```diff
@@ -75,18 +75,16 @@ public function store(Request $request, $sender_id, $recipient_id)
...
             abort(404);
         }
-        $conversation = DB::select('SELECT * FROM messages Where sender_id = ? AND recipient_id = ? OR sender_id = ? AND recipient_id = ?', [$sender_id, $recipient_id, $recipient_id, $sender_id]);
+        $conversation = DB::select('SELECT * FROM messages WHERE sender_id = ? AND recipient_id = ? OR sender_id = ? AND recipient_id = ?', [$sender_id, $recipient_id, $recipient_id, $sender_id]);
         $users = ['sender_id'    => $sender_id,
                   'recipient_id' => $recipient_id, ];
 
-        return view('pages.inbox')->withConversation($conversation)->withUsers($users);
+        $userInfo = User::find($recipient_id);
```
Comment (alexroden) [inline app/Http/Controllers/MessagesController.php]: Are we not able to bind the user in the route?

File: resources/views/pages/inbox.blade.php [hunk 3]
```diff
@@ -5,7 +5,7 @@
     <div class="jumbotron">
         
 
-        
+
         @if (! empty($conversation))
 
                     <h3 style="text-align: center;" class="display-4">Inbox with {{App\User::find($users['recipient_id'])->name}}</h3>
```
Comment (alexroden) [inline resources/views/pages/inbox.blade.php]: Ah!!! We should try to stay away from using classes in views

Comment (alexroden) [reply resources/views/pages/inbox.blade.php]: Tbh, you could map this to include the user.

Comment (alexroden) [reply resources/views/pages/inbox.blade.php]: Give us a shout when you get to this and I will show you what I mean

File: resources/views/pages/inbox.blade.php [hunk 4]
```diff
@@ -21,9 +21,12 @@
                     <small style="float: right;">Sent at {{Carbon::parse($convo->created_at)->format('d F Y H:i')}} <a href="/delete-message/{{$convo->id}}">Unsend</a></small>
                 </li>
         @else
-                
+
                 <li class="list-group-item" style="float:left">
-                    <h3>{{App\User::find($convo->sender_id)->name}}</h3>{{$convo->message}}<br>
+                    @if (isset($userInfo->image_path))
```
Comment (alexroden) [inline resources/views/pages/inbox.blade.php]: Wouldn't the `$userInfo->image_path` just return as `null` anyway? If it is on the model.

File: resources/views/pages/posts.blade.php [hunk 5]
```diff
@@ -8,7 +8,7 @@
                 <ul class="list-group">
                         <li class="list-group-item">
                                 @if (isset($post->user->image_path))
-                                <img style="width:40px;height:40px;"src="{{asset($post->user->image_path)}}" class="rounded">
+                                <a href="profile/{{$post->author_id}}"><img style="width:40px;height:40px;"src="{{asset($post->user->image_path)}}" class="rounded"></a>
```
Comment (alexroden) [inline resources/views/pages/posts.blade.php]: Can we start name routes, please?

**Comment by alex-bowers** [comment]
@maxsolomon14 Did you see Alex's comments above?

**Comment by maxsolomon14** [comment]
@alex-bowers I did, I'm currently implementing named routes

File: resources/views/pages/inbox.blade.php (hunk 4 above)
Comment (maxsolomon14) [reply resources/views/pages/inbox.blade.php]: @alexroden good point

**Pull request merged**